scripts/examples/gn_to_cmake.py

```diff
@@ -0,0 +1,760 @@
+#!/usr/bin/env python
+
+#
+#    Copyright (c) 2021-2021 Project CHIP Authors.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+##
+#    @file
+# Usage: gn_to_cmake.py <json_file_name>
```

Comment:
please explain what this does rather than usage example.

In my mind I am wondering: is this java specific (if so, it should probably not be in scripts/examples) or something else?

Reply:
I have tested by build example, it can build success now. 
My understanding, build_examples.py is a CLI tool which can build the apk with one command, while  gn_to_cmake.py will generate cmake files, so that we can build everything in Android studio, so that we can have better debuger in IDE. 
If any c/cpp code is not build in Android studio, android studio will not index them (as below image shows, the cpp code shows almost like a plain txt file, and we can not jump to functions, and we can not watch any values in content) 
![image](https://user-images.githubusercontent.com/1230620/130546804-6a3e8e32-26ed-41cb-b5ea-47690c987d0a.png)

With cmake, android studio now looks like below. Now we can see class defines ( sorry a little weak in my screenshot), can jump to function defines, and can watch content while debug)
![image](https://user-images.githubusercontent.com/1230620/130547652-4bd41546-9010-4373-b988-3beff2e72a57.png)

Also, this will help to dev in Microsoft windows, because everything is build in  android studio.

Reply:
Please explain in a comment inside the file. If the file name is not obvious, one should be able to open the file and understand its purpose.

The current name is 'gn_to_cmake' which tells me it seems to 'convert gn to cmake' which makes me think it converts 'BUILD.gn' to 'CMakeLists.txt' which is not the case here. Either rename it to be clearer (e.g. `gn_json_script_to_cmakelists.py` but that is quite long) or give an explanation here on what the script does (like taking the json data during a gn build and referencing https://gn.googlesource.com/gn/+/HEAD/docs/reference.md#generic-json-output for explanation)